aepsych/models/inducing_point_allocators.py

```diff
@@ -0,0 +1,303 @@
+        inputs: Optional[torch.Tensor] = None,
+        covar_module: Optional[torch.nn.Module] = None,
+        num_inducing: int = 10,
+        input_batch_shape: torch.Size = torch.Size([]),
+    ) -> torch.Tensor:
+        """
+        Generates `num_inducing` inducing points within the specified bounds using Sobol sampling.
+
+        Args:
+            inputs (torch.Tensor): Input tensor, not required for Sobol sampling.
+            covar_module (torch.nn.Module, optional): Kernel covariance module; included for API compatibility, but not used here.
+            num_inducing (int, optional): The number of inducing points to generate. Defaults to 10.
+            input_batch_shape (torch.Size, optional): Batch shape, defaults to an empty size; included for API compatibility, but not used here.
+
+
+        Returns:
+            torch.Tensor: A (num_inducing, d)-dimensional tensor of inducing points within the specified bounds.
+
+        Raises:
+            ValueError: If `bounds` is not provided.
+        """
+
+        # Validate bounds shape
+        assert (
+            self.bounds.shape[0] == 2
+        ), "Bounds must have shape (2, d) for Sobol sampling."
+
+        # Generate Sobol samples within the unit cube [0,1]^d and rescale to [bounds[0], bounds[1]]
+        inducing_points = draw_sobol_samples(
+            bounds=self.bounds, n=num_inducing, q=1
+        ).squeeze()
+
+        # Ensure correct shape in case Sobol sampling returns a 1D tensor
+        if inducing_points.ndim == 1:
+            inducing_points = inducing_points.view(-1, 1)
+
+        return inducing_points
+
+    @classmethod
+    def from_config(
+        cls,
+        config: Config,
+        name: Optional[str] = None,
+        options: Optional[Dict[str, Any]] = None,
+    ) -> "SobolAllocator":
+        """Initialize a SobolAllocator from a configuration object.
+
+        Args:
+            config (Config): Configuration object.
+            name (str, optional): Name of the allocator, defaults to None.
+            options (Dict[str, Any], optional): Additional options, defaults to None.
+
+        Returns:
+            SobolAllocator: A SobolAllocator instance.
+        """
+        return cls(**cls.get_config_options(config, name, options))
+
+    @classmethod
+    def get_config_options(
+        cls,
+        config: Config,
+        name: Optional[str] = None,
+        options: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """Get configuration options for the SobolAllocator.
+
+        Args:
+            config (Config): Configuration object.
+            name (str, optional): Name of the allocator, defaults to None.
+            options (Dict[str, Any], optional): Additional options, defaults to None.
+
+        Returns:
+            Dict[str, Any]: Configuration options for the SobolAllocator.
+        """
+        if name is None:
+            name = cls.__name__
+        bounds = config.gettensor(name, "bounds")
```

Comment:
I think we usually don't have bounds explicitly in the config, so we should construct them from lb/ub.

Reply:
I think a utility function went in recently that'll produce 2D bounds. It should be in the root utils.py

Reply:
Still a problem. Find the get_bounds function in transform and use it here. Ensure a test covers any case where this would break if get_bounds moved.